src/modules/musig/musig.md

```diff
@@ -0,0 +1,196 @@
+for this data structure in `include/secp256k1_musig.h`.
+
+## Key Generation
+
+To use MuSig, users must first compute their combined public key `P`, which is
+suitable for use on a blockchain or other public key repository. They do this
+by calling `secp256k1_musig_pubkey_combine`.
+
+This function takes as input a list of public keys `P_i` in the argument
+`pubkeys`. It outputs the combined public key `P` in the out-pointer `combined_pk`
+and hash `L` in the out-pointer `pk_hash32`, if this pointer is non-NULL.
+
+## Signing
+
+A participant who wishes to sign a message (as opposed to observing/auditing the
+signature process, which is also a supported mode) acts as follows.
+
+### Signing Participant
+
+1. The signer starts the session by calling `secp256k1_musig_session_initialize`.
+   This function outputs
+   - an initialized session state in the out-pointer `session`
+   - an array of initialized signer data in the out-pointer `signers`
+   - a commitment `H(R_i)` to a nonce in the out-pointer `nonce_commitment32`
+   It takes as input
+   - a unique session ID `session_id32`
+   - (optionally) a message to be signed `msg32`
+   - the combined public key output from `secp256k1_musig_pubkey_combine`
+   - the public key hash output from `secp256k1_musig_pubkey_combine`
+   - the signer's index `i` `my_index`
+   - the signer's secret key `seckey`
+2. The signer then communicates `H(R_i)` to all other signers, and receives
+   commitments `H(R_j)` from all other signers `j`. These hashes are simply
+   length-32 byte arrays which can be communicated however is communicated.
+3. Once all signers nonce commitments have been received, the signer records
+   these commitments with the function `secp256k1_musig_session_get_public_nonce`.
+   This function updates in place
+   - the session state `session`
+   - the array of signer data `signers`
+   taking in as input the list of commitments `commitments` and outputting the
+   signer's public nonce `R_i` in the out-pointer `nonce`.
+4. The signer then communicates `R_i` to all other signers, and receives `R_j`
+   from each signer `j`. On receipt of a nonce `R_j` he calls the function
+   `secp256k1_musig_set_nonce` to record this fact. This function checks that
+   the received nonce is consistent with the previously-received nonce and will
+   return 0 in this case. The signer must also call this function with his own
+   nonce and his own index `i`.
+   These nonces `R_i` are secp256k1 public keys; they should be serialized using
+   `secp256k1_ec_pubkey_serialize` and parsed with `secp256k1_ec_pubkey_parse`.
+5. Once all nonces have been exchanged in this way, signers are able to compute
+   their partial signatures. They do so by calling `secp256k1_musig_session_combine_nonces`
+   which updates in place
+   - the session state `session`
+   - the array of signer data `signers`
+   It outputs an auxilary integer `nonce_is_negated` and has an auxilary input
```

Comment:
s/auxilary/auxiliary/g

Reply:
:( I think I accidentally added the wrong spelling to my spellchecker at some point.